src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.cs

```diff
@@ -95,62 +143,72 @@ internal CacheEntry(object key, MemoryCache memoryCache)
         /// Gets or sets the priority for keeping the cache entry in the cache during a
         /// memory pressure triggered cleanup. The default is <see cref="CacheItemPriority.Normal"/>.
         /// </summary>
-        public CacheItemPriority Priority { get => _state.Priority; set => _state.Priority = value; }
+        public CacheItemPriority Priority { get => (CacheItemPriority)_priority; set => _priority = (byte)value; }
 
-        /// <summary>
-        /// Gets or sets the size of the cache entry value.
-        /// </summary>
-        public long? Size
+        internal long Size => _size;
+
+        long? ICacheEntry.Size
         {
-            get => _size;
+            get => _size < 0 ? null : _size;
             set
             {
                 if (value < 0)
                 {
                     throw new ArgumentOutOfRangeException(nameof(value), value, $"{nameof(value)} must be non-negative.");
                 }
 
-                _size = value;
+                // disallow entry size changes after it has been committed
+                if (_isDisposed)
```

Comment:
Why the behavior change here?

Reply:
From the PR description: Added a guard against ICacheEntry.Size changes after it has been committed to cache.

Previously it was possible to commit a CacheEntry with on size, change the size and then remove it causing the cacheSize value to be permanently off (could end up even negative).

Reply:
Is there a test covering this use case?